examples/audio/main/bsp_audio_example.c

```diff
@@ -59,77 +49,27 @@ typedef struct __attribute__((packed))
-        if (ret == ESP_FAIL) {
-            ESP_LOGE(TAG, "Failed to mount or format filesystem");
-        } else if (ret == ESP_ERR_NOT_FOUND) {
-            ESP_LOGE(TAG, "Failed to find SPIFFS partition");
-        } else {
-            ESP_LOGE(TAG, "Failed to initialize SPIFFS (%s)", esp_err_to_name(ret));
-        }
-
-        return ret;
-    }
-
-    size_t total = 0, used = 0;
-    ret = esp_spiffs_info(NULL, &total, &used);
-
-    if (ret != ESP_OK) {
-        ESP_LOGE(TAG, "Failed to get SPIFFS partition information (%s)", esp_err_to_name(ret));
-    } else {
-        ESP_LOGI(TAG, "Partition size: total: %d, used: %d", total, used);
-    }
-
-    return ret;
-}
-
 static void audio_task(void *arg)
 {
-    /* Create and configure ES8311 I2C driver */
-    es8311_handle_t es8311_dev = es8311_create(BSP_I2C_NUM, ES8311_ADDRRES_0);
-    const es8311_clock_config_t clk_cfg = BSP_ES8311_SCLK_CONFIG(SAMPLE_RATE);
-    es8311_init(es8311_dev, &clk_cfg, ES8311_RESOLUTION_16, ES8311_RESOLUTION_16);
-    es8311_voice_volume_set(es8311_dev, DEFAULT_VOLUME, NULL);
-
-    /* Microphone settings */
-    es7210_dev_handle_t es7210_handle = NULL;
-    const es7210_i2c_config_t es7210_i2c = BSP_ES7210_I2C_CONFIG(BSP_I2C_NUM);
-    ESP_ERROR_CHECK(es7210_new_codec(&es7210_i2c, &es7210_handle));
-    const es7210_codec_config_t codec_conf = BSP_ES7210_CONFIG(22050, I2S_MCLK_MULTIPLE_512);
-    ESP_ERROR_CHECK(es7210_config_codec(es7210_handle, &codec_conf));
-    ESP_ERROR_CHECK(es7210_config_volume(es7210_handle, 32));
-
-    /* Configure I2S peripheral and Power Amplifier */
-    bsp_audio_init(NULL, &i2s_tx_chan, &i2s_rx_chan);
-    bsp_audio_poweramp_enable(true);
+    esp_codec_dev_handle_t spk_codec_dev = bsp_audio_codec_speaker_init();
+    assert(spk_codec_dev);
+    esp_codec_dev_set_out_vol(spk_codec_dev, DEFAULT_VOLUME);
+    esp_codec_dev_handle_t mic_codec_dev = bsp_audio_codec_microphone_init();
 
     /* Pointer to a file that is going to be played */
-    const char music_filename[] = "/spiffs/16bit_mono_22_05khz.wav";
-    const char recording_filename[] = "/spiffs/recording.wav";
+    const char music_filename[] = BSP_SPIFFS_MOUNT_POINT"/16bit_mono_22_05khz.wav";
+    const char recording_filename[] = BSP_SPIFFS_MOUNT_POINT"/recording.wav";
     const char *play_filename = music_filename;
 
     while (1) {
         uint8_t btn_index = 0;
         if (xQueueReceive(audio_button_q, &btn_index, portMAX_DELAY) == pdTRUE) {
-            btn_index++; /* The first button is not handled here */
             switch (btn_index) {
             case BSP_BUTTON_REC: {
+                if (mic_codec_dev == NULL) {
+                    ESP_LOGW(TAG, "This board does not support microphone recording!");
+                    break;
+                }
                 /* Writing to SPIFFS from internal RAM is ~15% faster than from external SPI RAM */
                 int16_t *recording_buffer = heap_caps_malloc(BUFFER_SIZE, MALLOC_CAP_INTERNAL);
```

Comment:
We cannot use `MALLOC_CAP_INTERNAL` with LyraT (ESP32), it crashing when we want to fwrite into SPIFFS. The `MALLOC_CAP_DEFAULT` is working properly.